plugins/containers-plugin/src/containers-tree-data-provider.ts

```diff
@@ -267,3 +270,27 @@ export class ContainersTreeDataProvider implements theia.TreeDataProvider<ITreeN
         }
     }
 }
+
+export const CONTAINERS_PLUGIN_RUN_TASK_COMMAND_ID = 'containers-plugin-run-task';
+/**
+ * Command handler which is invoked when a user clicks on a task in the Workspace panel.
+ * This is needed if a command doesn't have container to be run in specified.
+ * In such case we run the command in the container under which this command was clicked.
+ */
+export async function containersTreeTaskLauncherCommandHandler(source: string, label: string, containerName: string): Promise<void> {
+    const tasks: theia.Task[] = await theia.tasks.fetchTasks({ type: 'che' });
+    for (const task of tasks) {
+        if (task.name === label && task.source === source) {
+            if (!task.definition.target) {
+                task.definition.target = {};
+            }
+            task.definition.target.containerName = containerName;
+
```

Comment:
```
task.definition.target = { containerName };
```
Is better instead of 
```
           if (!task.definition.target) {
                task.definition.target = {};
            }
            task.definition.target.containerName = containerName;
```
WDYT?

Reply:
We cannot do that because we'll lose other properties which are stored under `target` key if any.

Reply:
ok agree